service/src/main/java/uk/gov/hmcts/reform/fpl/service/DirectionHelperService.java

```diff
@@ -77,21 +82,122 @@
+                    .response(element.getValue().getResponses().stream()
+                        .filter(response -> response.getValue().getDirectionId().equals(element.getId()))
+                        .filter(response -> response.getValue().getAssignee().getValue().equals(assignee))
+                        .map(Element::getValue)
+                        .findFirst()
+                        .orElse(null))
+                    .build())
+                .build())
+            .collect(toList());
+    }
+
+    /**
+     * Collects directions for all parties and places them into single map, where the key is the role direction,
+     * and the value is the list of directions associated.
+     *
+     * @param caseData data from case.
+     * @return Map of roles and directions.
+     */
+    public Map<String, List<Element<Direction>>> collectDirectionsToMap(CaseData caseData) {
+        final List<Element<Direction>> emptyList = ImmutableList.of();
+
+        return Map.of(
+            ALL_PARTIES.getValue(), defaultIfNull(caseData.getAllParties(), emptyList),
+            LOCAL_AUTHORITY.getValue(), defaultIfNull(caseData.getLocalAuthorityDirections(), emptyList),
+            CAFCASS.getValue(), defaultIfNull(caseData.getCafcassDirections(), emptyList),
+            COURT.getValue(), defaultIfNull(caseData.getCourtDirections(), emptyList),
+            PARENTS_AND_RESPONDENTS.getValue(), defaultIfNull(caseData.getParentsAndRespondentsDirections(), emptyList),
+            OTHERS.getValue(), defaultIfNull(caseData.getOtherPartiesDirections(), emptyList));
+    }
+
+    /**
+     * Adds directionId and assignee wherever a response is present within a direction.
+     *
+     * @param directionMap a map of directions where assignee is key and value is a list of directions.
+     * @return a list of directions with hidden variables added where a response is present.
+     */
+    public List<Element<Direction>> addHiddenVariablesToResponseForManyAssignees(
+        Map<String, List<Element<Direction>>> directionMap) {
+        return directionMap.entrySet()
+            .stream()
+            .map(entry -> addHiddenVariablesToResponseForAssignee(entry.getKey(), entry.getValue()))
+            .flatMap(List::stream)
+            .collect(toList());
+    }
+
+    private List<Element<Direction>> addHiddenVariablesToResponseForAssignee(String assignee,
+                                                                             List<Element<Direction>> directions) {
+        return directions.stream()
+            .filter(element -> isNotEmpty(element.getValue().getResponse()))
+            .filter(element -> isNotEmpty(element.getValue().getResponse().getComplied()))
+            .map(x -> Element.<Direction>builder()
+                .id(x.getId())
+                .value(x.getValue().toBuilder()
+                    .response(x.getValue().getResponse().toBuilder()
+                        .assignee(DirectionAssignee.fromString(assignee))
```

Comment:
do we mind if `assignee` is `null` here, given `DirectionAssignee.fromString(...)` could return `null`?

Reply:
DirectionAssignee could return null... however we always pass in the keys from our role collections, which will always be the values of DirectionAssignee. So I don't believe this will ever happen